hydra-node/src/Hydra/Network/Etcd.hs

```diff
@@ -229,7 +234,22 @@ withEtcdNetwork tracer protocolVersion config callback action = do
 
   httpUrl (Host h p) = "http://" <> toString h <> ":" <> show p
 
-  NetworkConfiguration{persistenceDir, listen, advertise, peers} = config
+  NetworkConfiguration{persistenceDir, listen, advertise, peers, useSystemEtcd} = config
+
+-- | Return the path of the etcd binary. Will either install it first, or just
+-- assume there is one available on the system path.
+getEtcdBinary :: FilePath -> Bool -> IO FilePath
```

Comment:
Should: avoid boolean blindness and make this `Bool` a type also to be used in the `Options` module.

For example: `data UseSystemEtcd = InstallEtcd | SystemEtcd`